packages/react/src/hooks/usePopoverPositioning.jsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+
+const isOffscreen = (menuBody, menuDirection, menuButton /* , flipped, offset */) => {
+  const buttonRect = menuButton.getBoundingClientRect();
+  const tooltipRect = menuBody.getBoundingClientRect();
+
+  const windowWidth = window.innerWidth || document.documentElement.clientWidth;
+  const windowHeight = window.innerHeight || document.documentElement.clientHeight;
+
+  const offTop = buttonRect.top - tooltipRect.height < 0;
+  const offLeft = buttonRect.left - tooltipRect.width < 0;
+  const offRight = buttonRect.right + tooltipRect.width > windowWidth;
+  const offBottom = buttonRect.bottom + tooltipRect.height > windowHeight;
+  const T = offTop ? 'top' : '';
+  const R = offRight ? 'right' : '';
+  const B = offBottom ? 'bottom' : '';
+  const L = offLeft ? 'left' : '';
+
+  return [T, R, B, L].filter(Boolean).join('-');
+};
+
+export const usePopoverPositioning = ({
```

Comment:
Could you add jsdoc comments for the props and what it does / how it fixes the positioning issue. This all looks good, but I can see it being difficult to figure out whats happening in the future

Reply:
@sstone2423 Lots of docs added in this (https://github.com/carbon-design-system/carbon-addons-iot-react/pull/2052/commits/aa7e3eaf1b7a26dea4dc6dabe961c26a69f91059) newest commit.